docs/api/dirmonitor.lua

```diff
@@ -45,10 +45,15 @@ function dirmonitor:unwatch(fd_or_path) end
 ---edited, removed or added. A file descriptor will be passed to the
 ---callback in "multiple" mode or a path in "single" mode.
 ---
+---If an error occured, the error callback will be called with the error.
+---This function must not do anything more than saving the error somewhere else,
+---as it runs within a critical section and thus is prone to deadlocks.
+---
```

Comment:
```suggestion
---If an error occurred during the callback execution, the error callback will be called with the error object.
---The error callback must *not* raise errors, as they will result in deadlocks.
---
```
This suggestion needs formatting.

Reply:
The error callback can raise errors, in this case the error message is lost. It should never manipulate a coroutine or do long-running tasks.

Reply:
I see, I thought throwing errors was the problem.
Then something like:
```suggestion
---If an error occurred during the callback execution, the error callback will be called with the error object.
---The error callback must return as soon as possible, and must not handle coroutines as that might result in deadlocks.
---
```